src/pudl/transform/epacems.py

```diff
@@ -208,9 +230,9 @@ def transform(raw_df: pd.DataFrame, pudl_engine: sa.engine.Engine) -> pd.DataFra
     """
     return (
         raw_df.fillna({"gross_load_mw": 0.0, "heat_content_mmbtu": 0.0})
-        .pipe(harmonize_eia_epa_orispl)
+        .pipe(remove_leading_zeros_from_numeric_strings, "unit_id_epa")
+        .pipe(harmonize_eia_epa_orispl, pudl_engine)
         .pipe(fix_up_dates, plant_utc_offset=_load_plant_utc_offset(pudl_engine))
-        .pipe(add_facility_id_unit_id_epa)
```

Comment:
Can we hunt down the true meanings of these columns (by talking to EPA through @grgmiller if need be) and document them / name them appropriately rather than removing them? We've definitely been burned by removing stuff we don't understand in the past.

Alternatively it may make more sense to integrate them into the crosswalk table (maybe they're already there under a different name?), or into the plants/generators/boilers table, since they're probably actually determined by the the plant / generator / boiler / smokestack IDs anyway and don't need to be repeated a billion times here in the data table, when they can be included in the well normalized entity tables instead.

[A bit of searching suggests](https://enviro.epa.gov/enviro/EF_METADATA_HTML.ef_metadata_page?p_column_name=FACILITY_ID&p_table_name=AA_CEMS_DETAILS&p_topic=) that `facility_id` in the CEMS context refers to the EPA Greenhouse Gas Reporting Program ID. Maybe rename it to something more legible like `ghgrp_facility_id_epa`.

Renaming the `unitid` to `unit_id_epa` and removing the column previously named `unit_id_epa` also seems like it could result in confusion, but it sounds like the metadata was previously incorrect, and the renaming makes the `unit_id_epa` actually point at the thing the metadata was describing?  "Unit ID" and "Facility ID" are so ridiculously generic. If we're swapping column names around can we give it something more descriptive and unique? e.g. `emissions_unit_id_epa`?

Reply:
> Can we hunt down the true meanings of these columns (by talking to EPA through @grgmiller if need be)

I asked Greg about them and he wasn't sure, but I'll be darned if you don't have the magic touch for finding stuff like this!! I too would prefer to keep rather than remove records.

> Alternatively it may make more sense to integrate them into the crosswalk table (maybe they're already there under a different name?),

I already compared these values to those in the crosswalk and found no meaningful overlap. I can do some grouping to see if they at least correlate with one of the other fields (plant or unit). I'll also compare them to the ghg ids you found.

> or into the plants/generators/boilers table, since they're probably actually determined by the the plant / generator / boiler / smokestack IDs anyway

Do you mean putting these EPA plant / unit labels into the EIA entity tables? Or I suppose we could fabricate another CEMS table that just had these ids in it and not all the hourly info)

> If we're swapping column names around can we give it something more descriptive and unique? e.g. emissions_unit_id_epa?

I like that idea!

Reply:
Gocha I agree. I will see what I can do!

Reply:
Made an issue for this #1772